pkg/operation/botanist/monitoring.go

```diff
@@ -162,28 +162,30 @@ func (b *Botanist) DeploySeedMonitoring(ctx context.Context) error {
 		return err
 	}
 
-	if err := b.deployGrafanaCharts("users", usersDashboards.String(), basicAuthUsers, common.GrafanaUsersPrefix); err != nil {
-		return err
+	if b.Shoot.Info.Spec.Monitoring.Visualization.Enabled {
+		if err := b.deployGrafanaCharts("users", usersDashboards.String(), basicAuthUsers, common.GrafanaUsersPrefix); err != nil {
+			return err
+		}
 	}
 
 	// Check if we want to deploy an alertmanager into the shoot namespace.
 	if b.Shoot.WantsAlertmanager {
 		var (
 			alertingSMTPKeys = b.GetSecretKeysOfRole(common.GardenRoleAlertingSMTP)
 			emailConfigs     = []map[string]interface{}{}
-			to, _            = b.Shoot.Info.Annotations[v1alpha1constants.AnnotationShootOperatedBy]
 		)
-
-		for _, key := range alertingSMTPKeys {
-			secret := b.Secrets[key]
-			emailConfigs = append(emailConfigs, map[string]interface{}{
-				"to":            to,
-				"from":          string(secret.Data["from"]),
-				"smarthost":     string(secret.Data["smarthost"]),
-				"auth_username": string(secret.Data["auth_username"]),
-				"auth_identity": string(secret.Data["auth_identity"]),
-				"auth_password": string(secret.Data["auth_password"]),
-			})
+		for _, email := range b.Shoot.Info.Spec.Monitoring.Alerting.EmailReceivers {
```

Comment:
Again, checking `nil` would be recommended.

Reply:
if `WantsAlertmanager` is true then `EmailReceivers` cannot be nil. Should I still check regardless?

Reply:
It would probably be safer. If we some day touch this other function and forget to adapt it here we‘ll run into problems.